CMakeLists.txt

```diff
@@ -804,17 +814,22 @@ add_custom_target(check-install-via-properties-with-debug
 
     DEPENDS accp-jar tests-jar)
 
-add_custom_target(check
-    DEPENDS check-recursive-init
+set(check_targets check-recursive-init
     check-install-via-properties
     check-install-via-properties-with-debug
     check-junit
     check-junit-SecurityManager
     check-external-lib
     check-junit-AesLazy
     check-junit-AesEager
-    check-junit-DifferentTempDir
-)
+    check-junit-DifferentTempDir)
+
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  set(check_targets ${check_targets} check-with-jni-flag)
+endif()
+# For MacOS, check-with-jni-flag is not added due to failures in java.net.Inet6AddressImpl.lookupAllHostAddr: https://bugs.openjdk.org/browse/JDK-8205076
```

Comment:
Can you give a little more context why this exclusion? I thought I fixed JDK-8205076 for all platforms in 17.0.11. Do you need that fix in 11u?

Reply:
Yeah, it fails in current 11u. 11u backport is here: https://github.com/openjdk/jdk11u-dev/pull/2769 -- if approved, would be in October release. Does not block this PR, but maybe mention this check can be removed after 11u catches up?

Reply:
> The failure also happens for JDK 8

This is really weird, because this issue should not affect JDK 8. Looking into the raw logs, I see that tests look to be running with 11u still? Note the `java.base@11.0.23` below:

```
2024-06-13T15:07:34.5982930Z Complete job name: JDK8
...
2024-06-13T15:16:25.0335520Z FATAL ERROR in native method: Field type (static) mismatch in JNI get/set field operations
2024-06-13T15:16:25.0337710Z make[2]: *** [CMakeFiles/check-with-jni-flag.dir/all] Error 2
2024-06-13T15:16:25.0338320Z 	at java.net.Inet6AddressImpl.lookupAllHostAddr(java.base@11.0.23/Native Method)
2024-06-13T15:16:25.0339400Z make[1]: *** [CMakeFiles/coverage.dir/rule] Error 2
2024-06-13T15:16:25.0339970Z 	at java.net.InetAddress$PlatformNameService.lookupAllHostAddr(java.base@11.0.23/InetAddress.java:930)
2024-06-13T15:16:25.0342430Z make: *** [coverage] Error 2
2024-06-13T15:16:25.0345990Z 	at java.net.InetAddress.getAddressesFromNameService(java.base@11.0.23/InetAddress.java:1543)
2024-06-13T15:16:25.0349340Z 
```

Reply:
You're right. It uses JDK11 still for coverage target.